go/trace/trace.go

```diff
@@ -189,7 +189,7 @@ func StartTracing(serviceName string) io.Closer {
 }
 
 func fail(serviceName string) io.Closer {
-	options := make([]string, len(tracingBackendFactories))
+	options := make([]string, 0, len(tracingBackendFactories))
```

Comment:
This is a good catch. The bug is that we are allocating a slice with empty strings and then appending to it, so we end up with a slice that is twice as long as it really needs to be. Another way of fixing it would have been to set the already allocated elements instead of appending to the slice, but this is cleaner.